src/esm_parser/esm_parser.py

```diff
@@ -201,13 +223,22 @@ def _filter_variable(value):
             yielded=True,
         )
         for result in search_results:
-            print(result)
-            variables.append(
-                ConfigVariable(
-                    name=result[0].split(variable_separator)[-1],
-                    address=ConfigAddress(result[0]),
-                    should_resolve_from=result[1],
-                    is_resolved=False,
+            logger.debug(f"Full result: {result}")
+            re_pattern = re.escape(variable_start) + r"(.*?)" + re.escape(variable_end)
+            logger.debug(rf"Pattern to search with: {re_pattern}")
+            logger.debug(f"Result: {result[1]}")
+            regex_results = re.findall(re_pattern, result[1])
+            logger.debug(regex_results)
+            for regex_result in regex_results:
```

Comment:
It could be good idea to check if `regex_results` contain any data or not. If it is empty then an exception can be thrown.

Reply:
Possibly, yes. But, if it's empty, that just means you don't have any variables at all in the config you passed in, so the valid answer would be an empty list, which is what you get. I can write a test for that though